core/src/test/java/org/elasticsearch/action/bulk/BulkItemResponseTests.java

```diff
@@ -19,17 +19,149 @@
+import org.elasticsearch.common.xcontent.XContentType;
 import org.elasticsearch.test.ESTestCase;
 
+import java.io.IOException;
+import java.util.Map;
+
+import static org.elasticsearch.ElasticsearchExceptionTests.assertDeepEquals;
+import static org.elasticsearch.ElasticsearchExceptionTests.randomExceptions;
+import static org.elasticsearch.action.delete.DeleteResponseTests.assertDeleteResponse;
+import static org.elasticsearch.action.delete.DeleteResponseTests.randomDeleteResponse;
+import static org.elasticsearch.action.index.IndexResponseTests.assertIndexResponse;
+import static org.elasticsearch.action.index.IndexResponseTests.randomIndexResponse;
+import static org.elasticsearch.action.update.UpdateResponseTests.assertUpdateResponse;
+import static org.elasticsearch.action.update.UpdateResponseTests.randomUpdateResponse;
+import static org.elasticsearch.common.xcontent.XContentHelper.toXContent;
 import static org.hamcrest.Matchers.containsString;
 
 public class BulkItemResponseTests extends ESTestCase {
+
     public void testFailureToString() {
         Failure failure = new Failure("index", "type", "id", new RuntimeException("test"));
         String toString = failure.toString();
         assertThat(toString, containsString("\"type\":\"runtime_exception\""));
         assertThat(toString, containsString("\"reason\":\"test\""));
         assertThat(toString, containsString("\"status\":500"));
     }
+
+    public void testToAndFromXContent() throws IOException {
+        final XContentType xContentType = randomFrom(XContentType.values());
+
+        for (DocWriteRequest.OpType opType : DocWriteRequest.OpType.values()) {
+            int bulkItemId = randomIntBetween(0, 100);
+            boolean humanReadable = randomBoolean();
+            BytesReference originalBytes = null;
+            BulkItemResponse expectedBulkItemResponse = null;
+
+            if (opType == DocWriteRequest.OpType.INDEX || opType == DocWriteRequest.OpType.CREATE) {
+                expectedBulkItemResponse = new BulkItemResponse(bulkItemId, opType, randomIndexResponse());
+                originalBytes = toXContent(expectedBulkItemResponse, xContentType, humanReadable);
+
+            } else if (opType == DocWriteRequest.OpType.DELETE) {
+                expectedBulkItemResponse = new BulkItemResponse(bulkItemId, opType, randomDeleteResponse());
+                originalBytes = toXContent(expectedBulkItemResponse, xContentType, humanReadable);
+
+            } else if (opType == DocWriteRequest.OpType.UPDATE) {
+                Tuple<UpdateResponse, UpdateResponse> updates = randomUpdateResponse(xContentType);
+                expectedBulkItemResponse = new BulkItemResponse(bulkItemId, opType, updates.v2());
+                originalBytes = toXContent(new BulkItemResponse(bulkItemId, opType, updates.v1()), xContentType, humanReadable);
+
+            } else {
+                fail("Test does not support opType [" + opType + "]");
```

Comment:
Is this is necessary given we loop over OpType.values()? If other values are added in the future we should fail because expectedBulkItemResponse/originalBytes is not set anyway.